libs/strings/storage_string/src/lib.sw

```diff
@@ -0,0 +1,143 @@
+library;
+
+use std::{bytes::Bytes, storage::{clear_slice, get, get_slice, StorableSlice, store_slice}};
+use string::String;
+
+pub struct StorageString {}
+
+impl StorableSlice<String> for StorageString {
+    /// Takes a `String` type and saves the type in storage.
+    ///
+    /// ### Arguments
+    ///
+    /// * `string` - The string which will be stored.
+    ///
+    /// ### Number of Storage Accesses
+    ///
+    /// * Writes: `2`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     stored_string: StorageString = StorageString {}
+    /// }
+    ///
+    /// fn foo() {
+    ///     let mut string = String::new();
+    ///     string.push(5_u8);
+    ///     string.push(7_u8);
+    ///     string.push(9_u8);
+    ///
+    ///     storage.stored_string.store(string);
+    /// }
+    /// ```
+    #[storage(write)]
+    fn store(self, string: String) {
+        let key = __get_storage_key();
+        store_slice(key, string.as_raw_slice());
+    }
+
+    /// Constructs a `String` type from storage.
+    ///
+    /// ### Number of Storage Accesses
+    ///
+    /// * Reads: `2`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     stored_string: StorageString = StorageString {}
+    /// }
+    ///
+    /// fn foo() {
+    ///     let mut string = String::new();
+    ///     string.push(5_u8);
+    ///     string.push(7_u8);
+    ///     string.push(9_u8);
+    ///
+    ///     assert(storage.stored_string.load(key).is_none());
+    ///     storage.stored_string.store(string);
+    ///     let retrieved_string = storage.stored_string.load(key).unwrap();
+    ///     assert(string == retrieved_string);
+    /// }
+    /// ```
+    #[storage(read)]
+    fn load(self) -> Option<String> {
+        let key = __get_storage_key();
+        match get_slice(key) {
+            Option::Some(slice) => {
+                // Uncomment when https://github.com/FuelLabs/sway/issues/4408 is resolved
+                // Option::Some(String::from_raw_slice(slice))
+                Option::Some(String {
+                    bytes: Bytes::from_raw_slice(slice),
+                })
+            },
+            Option::None => Option::None,
+        }
+    }
+
+    /// Clears a stored string in storage.
+    ///
+    /// ### Number of Storage Accesses
+    ///
+    /// * Reads: `1`
+    /// * Clears: `2`
```

Comment:
There's a read, write and a clear? What is a clear? Is it a form of a write? I would use write.